packages/overlay/test/focus-trap.test.js

```diff
@@ -1,167 +1,76 @@
-      parent = fixtureSync('<overlay-components-wrapper></overlay-components-wrapper>');
-      await nextRender();
-      overlay = parent.$.defaultOverlay;
+          `;
+        }
+      };
       overlay.requestContentUpdate();
-      overlayPart = overlay.$.overlay;
-      focusableElements = getFocusableElements(overlay.$.overlay);
-      window.focus();
+      focusableElements = getFocusableElements(overlayPart);
     });
 
     afterEach(() => {
       overlay.opened = false;
     });
 
-    it('should not focus the overlay when focusTrap = false', async () => {
-      overlay.focusTrap = false;
-      overlay.opened = true;
-      await oneEvent(overlay, 'vaadin-overlay-open');
-      expect(getFocusedElementIndex()).to.be.eql(-1);
-    });
-
     it('should properly detect focusable elements inside the content', () => {
-      expect(focusableElements.length).to.equal(10);
+      expect(focusableElements.length).to.equal(5);
       expect(focusableElements[0]).to.eql(overlay.querySelector('textarea'));
       expect(focusableElements[1]).to.eql(overlay.querySelector('select'));
-      expect(focusableElements[2]).to.eql(overlay.$.overlay);
+      expect(focusableElements[2]).to.eql(overlayPart);
       expect(focusableElements[3]).to.eql(overlay.querySelector('button'));
-      expect(focusableElements[4]).to.eql(overlay.querySelector('vaadin-text-field').focusElement);
-      expect(focusableElements[5]).to.eql(overlay.querySelector('#text'));
-      expect(focusableElements[6]).to.eql(overlay.querySelector('#radioButton1').focusElement);
-      expect(focusableElements[7]).to.eql(overlay.querySelector('#radioButton2').focusElement);
-      expect(focusableElements[8]).to.eql(overlay.querySelector('#radioButton3').focusElement);
-      expect(focusableElements[9]).to.eql(overlay.querySelector('vaadin-button'));
+      expect(focusableElements[4]).to.eql(overlay.querySelector('input'));
     });
 
     it('should focus focusable elements inside the content when focusTrap = true', async () => {
       overlay.opened = true;
       await oneEvent(overlay, 'vaadin-overlay-open');
 
-      // TAB
+      // Tab
       for (let i = 0; i < focusableElements.length; i++) {
-        // Emulating control-state-mixin moving focus
-        if (focusableElements.filter(isElementFocused).length > 1) {
-          i += 1;
-        }
-
         const focusedIndex = getFocusedElementIndex();
         expect(focusedIndex).to.equal(i);
         tabKeyDown(focusableElements[focusedIndex]);
       }
       expect(getFocusedElementIndex()).to.eql(0);
 
-      // SHIFT+TAB
+      // Shift + Tab
       tabKeyDown(focusableElements[getFocusedElementIndex()], ['shift']);
-      for (let i = focusableElements.length - 1; i >= 0; i--) {
-        expect(getFocusedElementIndex()).to.eql(i);
-
-        // Emulating control-state-mixin moving focus
-        if (focusableElements.filter(isElementFocused).length > 1) {
-          i -= 1;
-        }
```

Comment:
We no longer use this mixin, and related logic does nothing since Vaadin 22. Previously, this check was true for e.g. `vaadin-text-field` as both the host element and the internal `<input>` were added to this array.